Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/app/app.component.html

```diff
@@ -1,18 +1,40 @@
 <body>
   <div>
     <h1> Elysia Range Simulator </h1>
-    <form name="simulator" #simulator="ngForm" novalidate>
+    <form name="simulator" [formGroup]="simulatorForm" #simulator="ngForm" novalidate>
       <label>Speed (km/h):</label>
       <br>
-      <input name="speed" type="number" placeholder="Speed" (keyup)="onKeySpeed($event)">
+      <input class="form-control" [ngClass]="{
+	  'invalid-field': inputSpeed.invalid && (inputSpeed.dirty || inputSpeed.touched),
+	  'valid-field': inputSpeed.valid && (inputSpeed.dirty || inputSpeed.touched)
+	  }" name="speed" type="number" formControlName="inputSpeed" placeholder="Speed" (keyup)="onKeySpeed($event)" >
```

Comment:
Lines 8-10 should have an indent, so it is easier to tell that it is part of the input field.